### PR TITLE
fix: quest faction preview and fields, closes #2678

### LINKED ISSUE #2678: [Bug] RewardFactions is completely wrong.
RewardFactions "value1" is a dbc lookup - not flat 7. 

Faction 47 is Ironforge, not venture company.

So the rewardfactions info is displaying wrong info for both amount and faction for all quests.

image

image

### PR DESCRIPTION
**Pull request by Helias**
closes https://github.com/azerothcore/Keira3/issues/2678

### CODE REVIEW

File: src/app/shared/modules/selectors/faction-selector/faction-selector-modal.component.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
   templateUrl: './faction-selector-modal.component.html',
-  styleUrls: ['./faction-selector-modal.component.scss'],
 })
 export class FactionSelectorModalComponent extends SearchSelectorModalComponent<Faction> {
   /* istanbul ignore next */ // because of: https://github.com/gotwarlost/istanbul/issues/690
-  constructor(protected bsModalRef: BsModalRef, public searchService: FactionSearchService) {
+  constructor(
+    protected bsModalRef: BsModalRef,
+    public searchService: FactionSearchService,
+  ) {
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/faction-selector/faction-selector-modal.component.ts:18]: same: better to use `inject`

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { BsModalService } from 'ngx-bootstrap/modal';
+
+import { BaseSelectorBtnComponent } from '../base-selector/base-selector-btn.component';
+import { QuestFactionSelectorModalComponent } from './quest-faction-selector-modal.component copy';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.ts:8]: we should try making new components with `standalone: true` and ideally migrate all the old ones to standalone too later on

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component copy.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+import { Faction, FACTION_SEARCH_FIELDS } from '@keira-types/faction.type';
+import { FactionSearchService } from '../../search/faction-search.service';
+import { SearchSelectorModalComponent } from '../base-selector/search-selector-modal.component';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component copy.ts:9]: same as above

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    searchService.query = '--mock query';
+
+    fixture = TestBed.createComponent(QuestFactionSelectorModalComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component.spec.ts:35]: same as above: setup function and what about actual tests?

File: src/app/shared/services/query/sqlite-query.service.ts [hunk 5]
```diff
@@ -14,7 +14,10 @@ export class SqliteQueryService extends BaseQueryService {
   private itemDisplayIdCache: Observable<string>[] = [];
   private spellDisplayIdCache: Observable<string>[] = [];
 
-  constructor(private sqliteService: SqliteService, private configService: ConfigService) {
+  constructor(
+    private sqliteService: SqliteService,
+    private configService: ConfigService,
+  ) {
```
Comment (FrancescoBorzi) [inline src/app/shared/services/query/sqlite-query.service.ts:20]: use `inject` and get rid of the constructor

File: src/app/shared/modules/selectors/faction-selector/faction-selector-modal.component.ts (hunk 1 above)
Comment (Helias) [reply src/app/shared/modules/selectors/faction-selector/faction-selector-modal.component.ts:18]: I do not know if I can do it considering the mother-class and the use of super() inside the constructor

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.ts (hunk 2 above)
Comment (Helias) [reply src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.ts:8]: build and tests fails, let's do this in another occasion/PR

File: src/app/shared/services/query/sqlite-query.service.ts (hunk 5 above)
Comment (Helias) [reply src/app/shared/services/query/sqlite-query.service.ts:20]: not sure if we can do this with super() inside the contrusctor and the mother class

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(QuestFactionSelectorBtnComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
[the remark this answers is not in the document]
Comment (Helias) [reply src/app/shared/modules/selectors/faction-selector/quest-faction-selector-btn.component.spec.ts:25]: this component has no content to verify, the view is a base html file tested in the base class

File: src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component.spec.ts (hunk 4 above)
Comment (Helias) [reply src/app/shared/modules/selectors/faction-selector/quest-faction-selector-modal.component.spec.ts:35]: this component has no content to verify, the view is a base html file tested in the base class

(I did the setup function)

**Pull request merged**